Improve documentation (#354)
stackitcloud/terraform-provider-stackit: README.md

[![Go Report Card](https://goreportcard.com/badge/github.com/stackitcloud/terraform-provider-stackit)](https://goreportcard.com/report/github.com/stackitcloud/terraform-provider-stackit) [![GitHub Release](https://img.shields.io/github/v/release/stackitcloud/terraform-provider-stackit)](https://registry.terraform.io/providers/stackitcloud/stackit/latest) ![GitHub go.mod Go version](https://img.shields.io/github/go-mod/go-version/stackitcloud/terraform-provider-stackit) [![GitHub License](https://img.shields.io/github/license/stackitcloud/terraform-provider-stackit)](https://www.apache.org/licenses/LICENSE-2.0)

# STACKIT Terraform Provider

> This project is the official [Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs) to integrate all the resources developed by [STACKIT](https://www.stackit.de/en/).

Reviewer: ```suggestion
This project is the official [Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs) for [STACKIT](https://www.stackit.de/en/), which allows you to manage STACKIT resources through Terraform.

```

Reviewer: I think something like this would be a bit clearer, not sure if it still fits the SEO criteria

Author: It still has the links and 3 key words STACKIT, Terraform, Provider :) So it looks good to me

        
## Getting Started

> To install [STACKIT Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs), copy and paste this code into your Terraform configuration. Then, run `terraform init`.

Reviewer: ```suggestion
To install the [STACKIT Terraform Provider](https://registry.terraform.io/providers/stackitcloud/stackit/latest/docs), copy and paste this code into your Terraform configuration. Then, run `terraform init`.

```

   }
   ```

### Key flow

    The following instructions assume that you have created a service account and assigned the necessary permissions to it, e.g. `project.owner`.

To use the key flow, you need to have a service account key, which must have an RSA key-pair attached to it.

> When creating the service account key, a new pair can be created automatically, which will be included in the service account key. This will make it much easier to configure the key flow authentication in the [STACKIT CLI](https://github.com/stackitcloud/stackit-cli/tree/main), by just providing the service account key.

Reviewer: Same here

> **Optionally**, you can provide your own private key when creating the service account key, which will then require you to also provide it explicitly to the [STACKIT CLI](https://github.com/stackitcloud/stackit-cli/tree/main), additionally to the service account key. Check the STACKIT Knowledge Base for an [example of how to create your own key-pair](https://docs.stackit.cloud/stackit/en/usage-of-the-service-account-keys-in-stackit-175112464.html#UsageoftheserviceaccountkeysinSTACKIT-CreatinganRSAkey-pair).

Reviewer: Oh, I think this should have been `provide it explicitly to the Terraform Provider` instead, not the CLI

  - The env var `TF_ACC_TEST_PROJECT_SERVICE_ACCOUNT_TOKEN` must be set as a valid token of the service account. Can also be set in the credentials file used by authentication (see [Authentication](#authentication) for more details)
  - The env var `TF_ACC_PROJECT_ID` is ignored
- For the Load Balancer service:
  - OpenStack credentials are required, as the acceptance tests use the OpenStack provider to set up the supporting infrastructure
    - These can be obtained after creating a user token through the [STACKIT Portal](https://portal.stackit.cloud/), in your project's Infrastructure API page
  - The env var `TF_ACC_OS_USER_DOMAIN_NAME` must be set as the OpenStack user domain name
  - The env var `TF_ACC_OS_USER_NAME` must be set as the OpenStack username
  - The env var `TF_ACC_OS_PASSWORD` must be set as the OpenStack password

**WARNING:** Acceptance tests will create real resources, which may incur in costs.

## Migration

For guidance on how to migrate to using this provider, please see our [Migration Guide](./MIGRATION.md).

## Reporting Issues

> If you encounter any issues or have suggestions for improvements, please open an issue in the [repository](https://github.com/SchwarzIT/terraform-provider-stackit/issues).

Reviewer 2: This is the link to the _other_ provider 👀. This is the correct one:
https://github.com/stackitcloud/terraform-provider-stackit/issues

Author: thx a lot :) changed the link
